crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -216,16 +242,76 @@ where
             inner,
             l2_block_source,
             metadata_storage,
+            da_source,
         };
         Ok(updater)
     }
+
+    fn validate_block_gas_capacity(
+        &self,
+        block_gas_capacity: u64,
+    ) -> anyhow::Result<NonZeroU64> {
+        NonZeroU64::new(block_gas_capacity)
+            .ok_or_else(|| anyhow!("Block gas capacity must be non-zero"))
+    }
+
+    async fn set_metadata(&mut self) -> anyhow::Result<()> {
+        self.metadata_storage
+            .set_metadata(self.inner.clone().into())
+            .map_err(|err| anyhow!(err))
+    }
+
+    async fn handle_normal_block(
+        &mut self,
+        height: u32,
+        gas_used: u64,
+        block_gas_capacity: u64,
+    ) -> anyhow::Result<()> {
+        let capacity = self.validate_block_gas_capacity(block_gas_capacity)?;
+
+        match &mut self.inner {
+            AlgorithmUpdater::V0(updater) => {
+                updater.update_l2_block_data(height, gas_used, capacity)?;
+            }
+            AlgorithmUpdater::V1(_) => {
+                return Err(anyhow!("V1 of the gas price algo has not been enabled yet"))
+                // TODO: update the DA record data with data received from the source
+                // updater.update_da_record_data(vec![])?;
```

Comment:
Is there an issue to address this? We should not leave any TODOs in the code without any plan for resolving them.

Reply:
this is still a part of https://github.com/FuelLabs/fuel-core/issues/2139. the PR has 1/2 to highlight the fact that 2/2 will resolve this

Reply:
Sure, but that is not visible for anyone reading the code when this is merged. In this scenario I'd prefer linking issue #2139 in the code to make it visible that this is intended to be resolved within the same issue, though in a different PR.

```suggestion
                // TODO(#2139): update the DA record data with data received from the source
                // updater.update_da_record_data(vec![])?;
```

Reply:
addressed in ea7e8a06f